pkg/controller.v1/tensorflow/pod.go

```diff
@@ -98,7 +99,25 @@ func (tc *TFController) reconcilePods(
 			if _, ok := spec.Template.Labels[TFPodGroupSettingLabel]; ok && CheckTFJobIsNotPending(tfjob) {
 				isWorker0Type := rtype == tfv1.TFReplicaTypeWorker && index == 0
 				if isWorker0Type {
-					worker0Completed = true
+					if value, ok := tfjob.Annotations[TFJobWaitingWorkerAnnotation]; ok && value == "true" {
+						if completedTime, ok := tfjob.Annotations["worker0-completed-time"]; !ok || completedTime == "" {
```

Comment:
I suggest you using the similar logic to determine if the TTL is reached instead of adding new annotation. And wrapper it to a function. 

```golang
currentTime := time.Now()
	ttl := tfJob.Spec.TTLSecondsAfterFinished
	if ttl == nil {
		// do nothing if the cleanup delay is not set
		return nil
	}
	duration := time.Second * time.Duration(*ttl)
	if currentTime.After(tfJob.Status.CompletionTime.Add(duration)) {
		
	}
```